Objects/frameobject.c

```diff
@@ -959,99 +969,154 @@ PyFrame_GetLineNumber(PyFrameObject *f)
     return PyUnstable_InterpreterFrame_GetLine(f->f_frame);
 }
 
+/*[clinic input]
+@critical_section
+@getter
+frame.f_lineno as frame_lineno
+
+Return the current line number in the frame.
+[clinic start generated code]*/
+
 static PyObject *
-frame_getlineno(PyObject *op, void *Py_UNUSED(closure))
+frame_lineno_get_impl(PyFrameObject *self)
+/*[clinic end generated code: output=70f35de5ac7ad630 input=87b9ec648b742936]*/
 {
-    PyFrameObject *f = PyFrameObject_CAST(op);
-    int lineno = PyFrame_GetLineNumber(f);
+    int lineno = PyFrame_GetLineNumber(self);
     if (lineno < 0) {
         Py_RETURN_NONE;
     }
-    else {
-        return PyLong_FromLong(lineno);
-    }
+    return PyLong_FromLong(lineno);
 }
 
+/*[clinic input]
+@getter
```

Comment:
I think critical section is not needed here as there is no corresponding setter for this and if the frame is running in a different thread then even adding critical section would not help. Is that right?